### PR TITLE
complete_substitutions_array used by ODT too

### PR DESCRIPTION
**Pull request by defrance**
Not possible to show the main call function between ODT and tab menu (function_call parameter used is the default...)
propose to add parameter function call for better tracability

### CODE REVIEW

File: htdocs/core/lib/functions.lib.php [hunk 1]
```diff
@@ -5349,11 +5349,15 @@ function complete_head_from_modules($conf,$langs,$object,&$head,&$h,$type,$mode=
 						if ($values[3]) $langs->load($values[3]);
 						if (preg_match('/SUBSTITUTION_([^_]+)/i',$values[2],$reg))
 						{
-							$substitutionarray=array();
-							complete_substitutions_array($substitutionarray,$langs,$object,array('needforkey'=>$values[2]));
-							$label=make_substitutions($reg[1], $substitutionarray);
+							$substitutionarray=head();
+							complete_substitutions_array($substitutionarray,$langs,$object,array('needforkey'=>$values[2]),'complete_head_from_modules');
+							if (count($substitutionarray)> 0 )
```
Comment (eldy) [inline htdocs/core/lib/functions.lib.php:5354]: I don't understand this. $label will not be defined.
$head should not be overwritten by $substitutionarray. substitutionarray is just an array that should contains 
"string key to replace" => "value to replace" to be used later by make_substitutions

File: htdocs/core/lib/functions.lib.php [hunk 2]
```diff
@@ -5349,11 +5349,15 @@ function complete_head_from_modules($conf,$langs,$object,&$head,&$h,$type,$mode=
 						if ($values[3]) $langs->load($values[3]);
 						if (preg_match('/SUBSTITUTION_([^_]+)/i',$values[2],$reg))
 						{
-							$substitutionarray=array();
-							complete_substitutions_array($substitutionarray,$langs,$object,array('needforkey'=>$values[2]));
-							$label=make_substitutions($reg[1], $substitutionarray);
+							$substitutionarray=head();
+							complete_substitutions_array($substitutionarray,$langs,$object,array('needforkey'=>$values[2]),'complete_head_from_modules');
```
Comment (eldy) [inline htdocs/core/lib/functions.lib.php:5353]: Why adding a parameter here ?
Default should be "completesubstitutionarray" as a the generic function. We should use this one in most cases so external module that bring this function will be called for most substitions.

Comment (defrance) [reply htdocs/core/lib/functions.lib.php:5353]: the complete_substitutions_arrayfunction is call in two places : 
- in odt for adding substitution elements (i use this feature on extraodt and customtabs to add some substitution
- in menu for change only the labels of a menu (i plan to use it to reordering tabs)

BUT without parameters it's the same function who is call in substitutions function (in my case extraodt_completesubstitutionarray ) 
with this parameters, I can select the good function to call in my substition file (/core/substitutions/functions_extraodt.lib.php)

With this code i keep the feature of label change but we need to change the function name
from
modulename_completesubstitutionarray 
to  
modulename_complete_head_from_modules
and i enhance it to made more change on menu tabs

I precise that the code is the same, just for the 2 case (5 ou 6 parameters)

**Comment by defrance** [comment]
finnaly find another way to to manage tabs

**Pull request closed without merging**